allofplos/transformations.py

```diff
@@ -23,28 +18,85 @@
 annotation = 'annotation'
 correction = 'correction'
 annotation_url = 'http://journals.plos.org/plosone/article/file?id=10.1371/annotation/'
-annotation_url_int = 'http://contentrepo.plos.org:8002/v1/objects/mogilefs-prod-repo?key=10.1371/annotation/'
 annotation_doi = '10.1371/annotation'
-BASE_URL_ARTICLE_LANDING_PAGE = 'http://journals.plos.org/plos{}/article?id='
+BASE_URL_ARTICLE_LANDING_PAGE = 'http://journals.plos.org/plos{}/article?id={}'
+BASE_URL_LANDING_PAGE = 'http://journals.plos.org/plos{}/'
+LANDING_PAGE_SUFFIX = '{}?id={}'
+
+plos_page_dict = {'article': 'article',
+                  'asset': 'article/asset',
+                  'articleFigsAndTables': 'article/assets/figsAndTables',
+                  'articleAuthors': 'article/authors',
+                  'citationDownloadPage': 'article/citation',
+                  'downloadBibtexCitation': 'article/citation/bibtex',
+                  'downloadRisCitation': 'article/citation/ris',
+                  'figuresPage': 'article/figures',
+                  'assetFile': 'article/file',
+                  'assetXMLFile': 'article/file',
+                  'articleMetrics': 'article/metrics',
+                  'articleRelated': 'article/related'}
 
 
-def filename_to_url(filename, plos_network=False):
+def get_base_page(journal):
+    """Make the base of a PLOS URL journal-specific.
+
+    Use in conjunction with `get_page()` in the Article class.
+    """
+    if len(journal.split(' ')) == 2:
+        BASE_LANDING_PAGE = BASE_URL_LANDING_PAGE.format(journal.lower().split(' ')[1])
+    elif 'negl' in journal.lower():
+        BASE_LANDING_PAGE = BASE_URL_LANDING_PAGE.format('ntds')
+    elif 'comp' in journal.lower():
+        BASE_LANDING_PAGE = BASE_URL_LANDING_PAGE.format('compbiol')
+    elif 'clinical trials' in journal.lower():
+        BASE_LANDING_PAGE = BASE_URL_LANDING_PAGE.format('ctr')
+    else:
+        print('URL error for {}'.format(journal))
+        BASE_LANDING_PAGE = BASE_URL_LANDING_PAGE.format('one')
+    return BASE_LANDING_PAGE
+
+
+def doi_to_journal(doi):
+    """For a given doi, get the PLOS journal that the article is published in.
+
+    For the subset of DOIs with 'annotation' in the name, defaults to PLOS ONE.
+    :return: string of journal name
+    """
+    journal = "PLOS ONE"
+    if 'pcbi' in doi:
```

Comment:
Hrm this seems like there could be a better way to write this… I’ll think about it but I’m thinking something like unpacking some kind of dict comprehension?

Reply:
like `plos_page_dict`? I'd be for that

Reply:
I think I know how I'd do it, if you'd like I can take a stab at it.